msgpack-core/src/main/java/org/msgpack/core/MessagePack.java

```diff
@@ -289,42 +312,112 @@ public MessageBufferPacker newBufferPacker()
         {
             return new MessageBufferPacker(this);
         }
-    }
 
-    /**
-     * MessageUnpacker configuration.
-     */
-    public static class UnpackerConfig
-    {
         /**
-         * Allow unpackBinaryHeader to read str format family  (default:true)
+         * Use String.getBytes() for converting Java Strings that are smaller than this threshold into UTF8.
+         * Note that this parameter is subject to change.
          */
-        public boolean allowReadingStringAsBinary = true;
+        public PackerConfig withSmallStringOptimizationThreshold(int bytes)
+        {
+            PackerConfig copy = clone();
+            copy.smallStringOptimizationThreshold = bytes;
+            return copy;
+        }
 
-        /**
-         * Allow unpackRawStringHeader and unpackString to read bin format family (default: true)
-         */
-        public boolean allowReadingBinaryAsString = true;
+        public int getSmallStringOptimizationThreshold()
+        {
+            return smallStringOptimizationThreshold;
+        }
 
         /**
-         * Action when encountered a malformed input
+         * When the next payload size exceeds this threshold, MessagePacker will call MessageBufferOutput.flush() before
+         * packing the data (default: 8192).
          */
-        public CodingErrorAction actionOnMalformedString = CodingErrorAction.REPLACE;
+        public PackerConfig withBufferFlushThreshold(int bytes)
+        {
+            PackerConfig copy = clone();
+            copy.bufferFlushThreshold = bytes;
+            return copy;
+        }
 
-        /**
-         * Action when an unmappable character is found
-         */
-        public CodingErrorAction actionOnUnmappableString = CodingErrorAction.REPLACE;
+        public int getBufferFlushThreshold()
+        {
+            return bufferFlushThreshold;
+        }
 
         /**
-         * unpackString size limit. (default: Integer.MAX_VALUE)
+         * When a packer is created with newPacker(OutputStream) or newPacker(WritableByteChannel), the stream will be
+         * buffered with this size of buffer (default: 8192).
          */
-        public int stringSizeLimit = Integer.MAX_VALUE;
+        public PackerConfig withBufferSize(int bytes)
+        {
+            PackerConfig copy = clone();
+            copy.bufferSize = bytes;
+            return copy;
+        }
+
+        public int getBufferSize()
+        {
+            return bufferSize;
+        }
+    }
+
+    /**
+     * MessageUnpacker configuration.
+     */
+    public static class UnpackerConfig
+            implements Cloneable
+    {
+        private boolean allowReadingStringAsBinary = true;
```

Comment:
Using getter is safer here because {Packer,Unpacker}Config are public API and change of the API means backward incompatibility. Exposing internal data structure seems negative for the future development.